qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.codeactions;
+
+import static com.redhat.qute.ls.commons.CodeActionFactory.createCommand;
+import static com.redhat.qute.utils.StringUtils.isSimilar;
+
+import java.text.MessageFormat;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+import com.redhat.qute.ls.commons.CodeActionFactory;
+import com.redhat.qute.ls.commons.client.ConfigurationItemEdit;
+import com.redhat.qute.ls.commons.client.ConfigurationItemEditType;
+import com.redhat.qute.parser.expression.Part;
+import com.redhat.qute.parser.template.Template;
+import com.redhat.qute.project.datamodel.JavaDataModelCache;
+import com.redhat.qute.services.commands.QuteClientCommandConstants;
+import com.redhat.qute.services.diagnostics.QuteErrorCode;
+import com.redhat.qute.settings.QuteValidationSettings.Severity;
+import com.redhat.qute.utils.QutePositionUtility;
+
+/**
+ * Abstract class for Qute code action.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public abstract class AbstractQuteCodeAction {
+
+	private static final String SET_IGNORE_SEVERITY_TITLE = "Ignore `{0}` problem.";
+
+	private static final String DID_YOU_MEAN_TITLE = "Did you mean `{0}`?";
+
+	protected final JavaDataModelCache javaCache;
+
+	public AbstractQuteCodeAction(JavaDataModelCache javaCache) {
+		this.javaCache = javaCache;
+	}
+
+	/**
+	 * Generate code action(s) sor similar values (Did you mean ...?)
+	 * 
+	 * @param part            the part node.
+	 * @param availableValues the available values.
+	 * @param template        the Qute template.
+	 * @param diagnostic      the diagnostic.
+	 * @param codeActions     list of CodeActions.s
+	 */

```

Comment:
Small typos:
* sor -> for
* The "s" after the "." for the param codeActions

Reply:
fixed